Dockerfile.base

```diff
@@ -0,0 +1,48 @@
+# build to ghcr.io/nilfoundation/build-base:$BOOST_VERSION
+
+ARG BOOST_VERSION=1.76.0
+ARG BOOST_VERSION_UNDERSCORED=1_76_0
+ARG BOOST_SETUP_DIR=/opt/boost_${BOOST_VERSION_UNDERSCORED}
+ARG BOOST_BUILD_DIRECTORY=/tmp/boost_${BOOST_VERSION_UNDERSCORED}
+
+FROM ubuntu:22.04 as boost_builder
+RUN DEBIAN_FRONTEND=noninteractive \
+    set -xe \
+    && apt-get update \
+    && apt-get -y --no-install-recommends --no-install-suggests install \
+        autoconf \
+        automake \
+        build-essential \
+        wget \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# using global args with their default versions
+ARG BOOST_VERSION
+ARG BOOST_VERSION_UNDERSCORED
+ARG BOOST_SETUP_DIR
+ARG BOOST_BUILD_DIRECTORY
+
+WORKDIR /tmp
+RUN set -xe \
+    && wget -q --no-check-certificate \
+      https://boostorg.jfrog.io/artifactory/main/release/${BOOST_VERSION}/source/boost_${BOOST_VERSION_UNDERSCORED}.tar.gz \
+    && mkdir ${BOOST_BUILD_DIRECTORY} \
+    && tar -xvf boost_${BOOST_VERSION_UNDERSCORED}.tar.gz \
+    && rm boost_${BOOST_VERSION_UNDERSCORED}.tar.gz
+
+WORKDIR ${BOOST_BUILD_DIRECTORY}
+RUN set -xe \
+    && sh ./bootstrap.sh --prefix=${BOOST_SETUP_DIR} \
```

Comment:
Do we need to support [Boost.Python](https://www.boost.org/doc/libs/1_82_0/libs/python/doc/html/index.html) library? 
I think it's not necessary for zkLLVM, but if we want to use it as base image, maybe in future for some projects we want to have it (--with-python=... for bootstrap).

Reply:
@Zerg1996 let's add it when we have a practical need for it.